### PR TITLE
initialize guess_atoms function PR

### PR DESCRIPTION
**Pull request by cadeduckworth**
- add helper function to correctly guess atom/element types from topology
- helper function `guess_atoms` used in `rdkit_conversion` function
- temporary changes pending MDAnalysis `guessers` class bug fixes

### CODE REVIEW

**Comment by cadeduckworth** [comment]
@orbeckst please review

I did not try it out on any cases outside of the notebook during development, however, I did incorporate usage into the dihedrals module and also added the test for the base module. Everything passed locally, so hopefully there are no issues with the CI tests.

If you see any issues please let me know. If everything looks good I will update my SAMPL9 environment, do a test run, and add documentation once we are sure everything is working as intended.

If you would like more detail on the new loop added for creating the boolean array used for identifying problem cases I can upload the updated notebook here.. just let me know!

File: mdpow/tests/test_workflows_base.py [hunk 1]
```diff
@@ -21,6 +23,19 @@ def molname_workflows_directory(tmp_path):
 
 class TestWorkflowsBase(object):
 
+    check_guessed_elements = np.array(['C', 'N', 'C', 'Cl', 'C'])
+
+    @pytest.fixture(scope='function')
+    def create_universe(self):
```
Comment (orbeckst) [inline mdpow/tests/test_workflows_base.py:29]: I call fixtures that provide a thing "thing" — the creation part is technical stuff.
```suggestion
    def universe(self):
```

File: mdpow/workflows/base.py [hunk 2]
```diff
@@ -173,3 +178,34 @@ def automated_project_analysis(project_paths, ensemble_analysis, **kwargs):
 
     logger.info('all analyses completed')
     return
+
+def guess_elements(universe_atom_group):
+    # input from dihedrals module
+    # universe_atom_group = universe.atoms
+    # universe from MDPOW FEP project results
+    names = universe_atom_group.names
+    masses = universe_atom_group.masses
```
Comment (orbeckst) [inline mdpow/workflows/base.py:187]: Normally I would say "we need to add checks that we can get masses etc" but within MDPOW we now that we use TPRs so I am fine with just *documenting* it.

File: mdpow/workflows/base.py [hunk 3]
```diff
@@ -173,3 +178,34 @@ def automated_project_analysis(project_paths, ensemble_analysis, **kwargs):
...
+    problem_cases = []
+    i=0
+    while i < len(masses):
+        if masses[i] != 0:
+            value = math.isclose(guessed_masses[i], masses[i], rel_tol=0.001)
+            value = not(value)
+        else:
+            value = False
+        problem_cases.append(value)
+        i+=1
```
Comment (orbeckst) [inline mdpow/workflows/base.py:201]: This is ugly. Can't we get `numpy.isclose()` get to work? And don't use `math` if we already have numpy.

If you provide me with a minimal example  where the old code didn't work, I'll have a look. (Basically, add a *test* that fails with the old numpy-based code)

File: mdpow/workflows/dihedrals.py [hunk 4]
```diff
@@ -181,11 +181,16 @@ def rdkit_conversion(u, resname):
 
     """
 
+    from .base import guess_elements
+    # temporary
+    # keep here to avoid circular imports?
+
```
Comment (orbeckst) [inline mdpow/workflows/dihedrals.py:187]: I currently don't see that base imports dihedrals in any way, or did I overlook something? What's your rationale for not putting the imports at the top of the file?

File: mdpow/tests/test_workflows_base.py [hunk 5]
```diff
@@ -7,42 +7,64 @@
...
+@pytest.fixture()
 def molname_workflows_directory(tmp_path):
     m = pybol.Manifest(str(MANIFEST))
     m.assemble('workflows', tmp_path)
     return tmp_path
 
 class TestWorkflowsBase(object):
 
-    @pytest.fixture(scope='function')
+    reference_guessed_elements = np.array(['C', 'N', 'C', 'Cl', 'C'])
```
Comment (orbeckst) [inline mdpow/tests/test_workflows_base.py:27]: The test failure is due to capitalization:
```suggestion
    reference_guessed_elements = np.array(['C', 'N', 'C', 'CL', 'C'])
```

**Comment by orbeckst** [comment]
This needs some debugging. Ideally you can put whatever kills it at the moment in a test case.

**Review by orbeckst — APPROVED** [review]

**Pull request merged**